src/components/Common/NumericInput.js

```diff
@@ -0,0 +1,92 @@
+import React, { Component } from 'react'
+import { InputField } from './InputField'
+import { VALIDATION_TYPES } from '../../utils/constants'
+import update from 'immutability-helper'
+import { observer } from 'mobx-react'
+import { countDecimalPlaces } from '../../utils/utils'
+
+const { VALID, INVALID } = VALIDATION_TYPES
+
+@observer
+export class NumericInput extends Component {
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      value: '',
+      validation: {
```

Comment:
Since `NumericInput` can only have one value, I think this can be flattened to `this.state = { value, pristine, valid}`

Reply:
done, thanks!